homeassistant/components/rainmachine/config_flow.py

```diff
@@ -57,12 +56,11 @@ def __init__(self):
 
     async def async_step_user(self, user_input=None):
         """Handle the start of the config flow."""
-
         if not user_input:
             return await self._show_form()
 
-        if user_input[CONF_IP_ADDRESS] in configured_instances(self.hass):
-            return await self._show_form({CONF_IP_ADDRESS: "identifier_exists"})
+        await self.async_set_unique_id(user_input[CONF_IP_ADDRESS])
```

Comment:
If we also update existing entries and set a unique_id on them in `async_setup_entry` if not set already, we can remove `configured_instances` function.

Reply:
Great feedback, @MartinHjelmare! I'll come back around with another PR.